### PR TITLE
Fixed agent scrolling

### PR DESCRIPTION
**Pull request by abist**
Added a couple lines to VS code source, which shouldn't affect it. This was the only way I could solve it.

Fixes https://github.com/Microsoft/azuredatastudio/issues/2064

### CODE REVIEW

**Comment by anthonydresser** [comment]
I'm going to go back to the previous time we had this issue, its incredibly unlikely that this is the right fix.

**Comment by abist** [comment]
@anthonydresser I had a solution before but we didn't take it. If you have a solution, please let me know or we'll always be blocked by this issue.

File: src/vs/base/parts/tree/browser/treeView.ts [hunk 1]
```diff
@@ -511,10 +512,19 @@ export class TreeView extends HeightMap {
 			vertical: (typeof context.options.verticalScrollMode !== 'undefined' ? context.options.verticalScrollMode : ScrollbarVisibility.Auto),
 			useShadows: context.options.useShadows
 		});
-		this.scrollableElement.onScroll((e) => {
-			this.render(e.scrollTop, e.height, e.scrollLeft, e.width, e.scrollWidth);
-			this._onDidScroll.fire();
-		});
+
+		// {{SQL CARBON EDIT}}
+		if (this.context.dataSource instanceof JobStepsViewDataSource) {
```
Comment (kburtram) [inline src/vs/base/parts/tree/browser/treeView.ts]: I don't think it makes sense for the VS Code TreeControl to take a dependency on the SQL Agent JobStepsViewDataSource class.

**Comment by abist** [comment]
@kburtram I might have to rewrite the whole page again using SlickGrid table instead since tree doesn't work here. Should I be aiming for December release?

**Comment by kburtram** [comment]
@abist please work on other issues and leave this active (it doesn't need to land in Dec.).  I'll try to take a look to understand the problem here.  One way to avoid having the tree reference SQL Agent classes would be to move the related logic into a virtual function and then create a SqlAgentTree subclass that contains the special-case logic.  In any case, let me take a closer look before making a recommendation here.

**Comment by abist** [comment]
@kburtram Sounds good. I actually know the problem, so I can perhaps point it out to you if you need a little context.

**Comment by abist** [comment]
Updated PR - https://github.com/Microsoft/azuredatastudio/pull/3398

**Pull request closed without merging**